src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1321,32 +1321,39 @@ private static object FoldUncheckedIntegralBinaryOperator(BinaryOperatorKind kin
                 switch (kind)
                 {
                     case BinaryOperatorKind.IntAddition:
+                    case BinaryOperatorKind.NIntAddition:
                         return valueLeft.Int32Value + valueRight.Int32Value;
```

Comment:
>valueLeft.Int32Value + valueRight.Int32Value; [](start = 31, length = 45)

What would happen if left is Int32.MaxValue and the right is 10? Is this going to be equivalent to the result of a non constant addition for the same values on a platform where nint size is 8 bytes?